### PR TITLE
Fix DebugType and Optimize values for JIT tests.

### PR DESCRIPTION
**Pull request by pgavlin**
These values were incorrect given the behavior of the test build.

### CODE REVIEW

**Comment by pgavlin** [comment]
@richardlford PTAL.

**Comment by pgavlin** [comment]
@dotnet-bot test this please

**Comment by mikedn** [comment]
Why exactly is Optimize set to false? Won't this disable JIT optimizations thus making JIT tests more or less useless?

**Comment by pgavlin** [comment]
> Won't this disable JIT optimizations thus making JIT tests more or less useless?

Yes, this will disable JIT optimizations. These tests still serve to ensure that non-optimized code generation is correct, and are often part of a set of tests that target (release, release w/ opts, debug, debug w/ opts) configurations. @BruceForstall can comment further.

**Comment by mikedn** [comment]
Yes but now we have both `straccess1_cs_d.csproj` and `straccess1_cs_r.csproj` with `Optimize = false`. The only difference between them is `DebugType = Full` vs. `DebugType = None` but AFAIK that does not affect JIT optimizations.

**Comment by pgavlin** [comment]
Hm, I had thought that `/debug:full` did affect code generation in order to allow a debugger to attach to and debug a running process, but some simple experiments indicate that you're correct. Can someone else on @dotnet/jit-contrib chime in?

**Comment by pgavlin** [comment]
@dotnet-bot test Ubuntu x64 Debug Build and Test

**Comment by pgavlin** [comment]
@mmitche: any idea why the OSX/Ubuntu Debug legs haven't retriggered?

**Comment by BruceForstall** [comment]
Code generation shouldn't be affected, but debug information generation should. We've traditionally had the full matrix of {optimized,non-optimized}x{debug,non-debug}. Not clear how useful these have all been, but we've had them.

**Comment by mikedn** [comment]
As far as I can tell Roslyn always sets `DebuggableAttribute.DebuggingModes.IgnoreSymbolStoreSequencePoints` no matter what /debug option is used. Instead of PDB sequence points it relies on inserting nops. So even if it would be useful to test JIT's handling of PDB's sequence points it appears that there's no way to do what with C# assemblies.

**Comment by mmitche** [comment]
@pgavlin It looks like it did, I see a run a few hours ago.

**Comment by pgavlin** [comment]
@mmitche link? When I click through to the failure above, I see a run that started 16 days ago.

**Comment by mmitche** [comment]
@pgavlin Looks like the Ubuntu failure is old but osx is recent.

**Comment by mmitche** [comment]
@dotnet-bot test Ubuntu x64 Debug Build and Test

**Comment by pgavlin** [comment]
The OSX Debug failure is also old. The release failure is new but is unrelated (failures are in PAL tests).

**Comment by mmitche** [comment]
@pgavlin Okay that is because it appears the leg is disabled.  Ahh yes, we disabled the debug runs for TP reasons.

**Comment by mmitche** [comment]
@pgavlin Disabled for the same reason

**Comment by pgavlin** [comment]
Okay. Should I close and reopen this, then, or is this effectively green?

**Comment by mmitche** [comment]
Assuming you're okay with the release failure, then you're good to go.

**Comment by pgavlin** [comment]
@dotnet-bot test OSX x64 Release Build and Test

**Comment by pgavlin** [comment]
@richardlford Ping.

**Comment by richardlford** [comment]
@pgavlin, I'm looking now.

**Comment by richardlford** [comment]
@pgavlin, looks reasonable. These corrections are based on the data from our json file?
Can you summarize your strategy in these changes? E.G. Empty Optimization tag -> False, etc.

**Comment by pgavlin** [comment]
> Can you summarize your strategy in these changes?

In general:
Empty tag -> No corresponding option specified on the original command line

For <DebugType>, the value is simply whatever was specified on the original command line.

For <Optimize>, the value is either True or False based on whether or not /o+ or /o- was specified on the original command line.

**Comment by AndyAyersMS** [comment]
I worry that "empty" is not really a viable choice. I have seen cases where the actual debuggable attribute in the assembly varies in ways I don't understand or expect (eg from one local build to the next, or from one machine to another).

**Comment by pgavlin** [comment]
The "empty" value essentially means that the original test compilation defaulted its debug type/optimize flag to the global settings for the test build. In our case, that means that these tests will build differently in a Debug build compared to a Release build. We could certainly make a deliberate choice here; what that choice should be is not something I'm sure of.

**Comment by CarolEidt** [comment]
@pgavlin There is some evidence that the "empty" value is non-deterministic - or at least that there are differences in the way tests are built. I was running into a Release test failure on the lab machine that I couldn't repro locally. The test on the lab machine had debug off for Main, but the one that I built (using the instructions for building Windows tests and copying them over) had it on. I would be in favor of running the full matrix - i.e. tests that are built both debug and opt, and run with a runtime/jit that is built both Debug and Release.

**Comment by AndyAyersMS** [comment]
I have seen this too -- which is why when I see all these test issues closed with "doesn't repro" I wonder if we're testing the same binaries.

**Comment by AndyAyersMS** [comment]
I think I tracked down at least one source of this mysterious variance. 

The managed code build uses the compilers that come with VS2015 (or VS2013, I suppose, depending on how you configure your build).

VS2015 and VS2015 update 1 produce different IL for the same test cases. I noticed this in `Jit\Regression\JitBlue\GitHub_2610` while trying to track down a different issue.

With VS2015, the Main method starts out...
```
  .method private hidebysig static int32
          Main() cil managed
  {
    .entrypoint
    // Code size       142 (0x8e)
    .maxstack  4
    .locals init (valuetype MyValue V_0,
             valuetype MyValue V_1,
             valuetype MyStruct V_2,
             bool V_3,
````

With VS2015 Update 1, it starts out...
```
  .method private hidebysig static int32 
          Main() cil managed
  {
    .entrypoint
    // Code size       147 (0x93)
    .maxstack  3
    .locals init (valuetype MyValue V_0,
             valuetype MyValue V_1,
             valuetype MyStruct V_2,
             valuetype MyStruct V_3,
             bool V_4,
```

Not 100% sure what version of VS the lab runs, but if you're not using what it uses, you may not be testing the exact same bits.
